client/shaders/fog.frag

```diff
@@ -15,6 +15,17 @@ void main() {
     vec3 view_pos = scaled_view_pos.xyz / scaled_view_pos.w;
     // Hyperbolic distance
     float dist = atanh(min(length(view_pos), 1));
-    // Exponential-squared fog
-    fog = vec4(0.5, 0.5, 0.5, exp(-pow(dist * fog_density, 2)));
+    // piecewise constant/exponential dropoff fog
+    float terrain_proportion;
+    float threshold = 1.0;
+    if (dist <= threshold)
+    {
+        terrain_proportion = 1.0;
+    } 
+    else 
+    {
+        terrain_proportion = exp(-(dist - threshold) * fog_density); 
+    }
```

Comment:
We should try to phrase this in a branch-free way, unless you can show that spirv-opt successfully translates it into one.

Reply:
I've undone the fog fade, so there is no longer branching.